### PR TITLE
PMS-23 created crud operations for patients

### PR DESCRIPTION
**Pull request by jaysingh321**

### CODE REVIEW

**Review by PraveenH18 — APPROVED** [review]

File: src/core/controllers/patientsController.js [hunk 1]
```diff
@@ -9,6 +10,143 @@ var getPatients = async (req, res) => {
   }
 };
 
+//Get patient by id
+const getPatientsbyid = async (req, res) => {
```
Comment (pallawir) [inline src/core/controllers/patientsController.js:14]: rename getPatientsbyid  to getPatientById

File: src/core/controllers/patientsController.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const models = require("../models/index");
 
-var getPatients = async (req, res) => {
+//Get all patients
+const getPatients = async (req, res) => {
   try {
     const patients = await models.Patients.findAll();
     res.json(patients);
```
Comment (pallawir) [inline src/core/controllers/patientsController.js:7]: Add check if patients is available then only perform further actions

File: src/core/controllers/patientsController.js [hunk 3]
```diff
@@ -9,6 +10,143 @@ var getPatients = async (req, res) => {
...
+    const {
+      hospitalId,
+      fullName,
+      age,
+      dateOfBirth,
+      email,
+      contactNumber,
+      nationality,
+    } = req.body;
+    const patient = await models.Patients.create({
```
Comment (pallawir) [inline src/core/controllers/patientsController.js:46]: Before creating add validation if all the required parameters are available

File: src/core/controllers/patientsController.js [hunk 4]
```diff
@@ -9,6 +10,143 @@ var getPatients = async (req, res) => {
...
+    res.status(201).json(patient);
+  } catch (error) {
+    res.status(500).json({ error: "Failed to create patient", error });
+  }
+};
+
+// delete patient
+const deletePatient = async (req, res) => {
+  try {
+    const { patientId } = req.query;
```
Comment (pallawir) [inline src/core/controllers/patientsController.js:64]: Add check to see if patient Id is available or not

File: src/core/controllers/patientsController.js [hunk 5]
```diff
@@ -9,6 +10,143 @@ var getPatients = async (req, res) => {
...
+    const {
+      hospitalId,
+      fullName,
+      age,
+      dateOfBirth,
+      email,
+      contactNumber,
+      nationality,
+    } = req.body;
+    const updatedPatient = await models.Patients.update(
```
Comment (pallawir) [inline src/core/controllers/patientsController.js:91]: Add check to see if patient Id is available or not

File: src/core/server.js [hunk 6]
```diff
@@ -6,11 +6,17 @@ const app = express();
...
 app.use(cors());
 
+app.use(express.json());
+
 const patientsContr = require("./controllers/patientsController");
 const port = 8080;
 
 app.get("/patients", patientsContr.getPatients);
-
+app.get("/patientsbyid", patientsContr.getPatientsbyid);
```
Comment (pallawir) [inline src/core/server.js:15]: Change patientsbyid to patientById

**Review by pallawir — APPROVED** [review]

**Pull request merged**